pull/github.go

```diff
@@ -149,30 +149,65 @@ func (ghc *GitHubContext) ChangedFiles() ([]*File, error) {
 
 func (ghc *GitHubContext) Commits() ([]*Commit, error) {
 	if ghc.commits == nil {
-		if err := ghc.loadPullRequestData(); err != nil {
-			return nil, err
+		totalCommits := ghc.pr.GetCommits()
+
+		var q struct {
+			Repository struct {
+				Object struct {
+					Commit struct {
+						History struct {
+							PageInfo v4PageInfo
+							Nodes    []*v4Commit
+						} `graphql:"history(first: $limit, after: $cursor)"`
+					} `graphql:"... on Commit"`
+				} `graphql:"object(oid: $oid)"`
+			} `graphql:"repository(owner: $owner, name: $name)"`
+		}
+		qvars := map[string]interface{}{
+			// always list commits from the head repository
+			// some commit details do not propagate from forks to the upstream
+			"owner": githubv4.String(ghc.pr.GetHead().GetRepo().GetOwner().GetLogin()),
+			"name":  githubv4.String(ghc.pr.GetHead().GetRepo().GetName()),
+			"oid":   githubv4.GitObjectID(ghc.pr.GetHead().GetSHA()),
+
+			"limit":  githubv4.Int(min(totalCommits, MaxPageSize)),
```

Comment:
Could we remove the `min` logic and just ask for `MaxPageSize` every time? This assumes that Github will return _at most_ `MaxPageSize` results and not error out.